index.js

```diff
@@ -0,0 +1,117 @@
+//module import
+const express = require('express');
+const helmet = require('helmet');
+const knex = require('knex');
+const bcrypt = require('bcryptjs')
+const session = require('express-session');
+const KnexSessionStore = require('connect-session-knex')(session)
+//knex config
+const dbConfig = require('./knexfile.js')
+const db = knex(dbConfig.development);
+//server config
+const server = express();
+
+// configure express-session middleware I'm not sure I like this so I'm doing to go the  session config route I now realize it's the same
+// server.use(
+//   session({
+//     name: 'login', // default is connect.sid
+//     secret: 'fly you fools!',
+//     cookie: {
+//       maxAge: 1 * 24 * 60 * 60 * 1000,
+//       secure: false, // only set cookies over https. Server will not send back a cookie over http.
+//     }, // 1 day in milliseconds
+//     httpOnly: true, // don't let JS code access cookies. Browser extensions run JS code on your browser!
+//     resave: false,
+//     saveUninitialized: false,
+//     store: new KnexSessionStore({
+//       tablename:"sessions",
+//       sidfieldname: 'sid',
+//       knex: db,
+//       createtable: true,
+//       clearInterval: 1000*60*60,
+//     })
+//   })
+// );
+
+//sessionConfig
+const sessionConfig = {
+  name: 'login',
+  secret: 'fly you fools',
+      cookie: {
+        maxAge: 1 * 24 * 60 * 60 * 1000,
+        secure: false, // only set cookies over https. Server will not send back a cookie over http.
+      },
+      resave:false,
+      saveUninitialized:false
+}
```

Comment:
What turned you off from using `KnexSessionStore`?

Reply:
Nothing huge I just didn't see it as being necessary. I guess I understand how the session is stored and cookies are passed. To me it'  more intuitive to just set the session configuration explicitly.